src/main/java/org/jboss/forge/arquillian/container/model/Container.java

```diff
@@ -142,12 +167,98 @@ public static String expandAbbr(String id) {
     }
 
     public static String abbr(String id) {
-        for (Map.Entry<String,String> abbr : ABBREVIATIONS.entrySet()) {
+        for (Map.Entry<String, String> abbr : ABBREVIATIONS.entrySet()) {
             if (id.contains(abbr.getValue())) {
                 id = id.replace(abbr.getValue(), abbr.getKey());
             }
         }
 
         return id;
     }
-}
\ No newline at end of file
+
+    public String getNameForChameleon() {
+        final String artifactId = getArtifactId();
+        if (Identifier.TOMCAT.getArtifactID().equals(artifactId)) {
+            return Identifier.TOMCAT.getName();
+        } else if (Identifier.JBOSS_AS.getArtifactID().equals(artifactId) && getName().contains(AS) && !getName().contains(AS_DOMAIN)) {
+            return Identifier.JBOSS_AS.getName();
+        } else if (Identifier.JBOSS_AS_DOMAIN.getArtifactID().equals(artifactId) && getName().contains(AS_DOMAIN)) {
+            return Identifier.JBOSS_AS_DOMAIN.getName();
+        } else if (Identifier.JBOSS_EAP_DOMAIN.getArtifactID().equals(artifactId) && getName().contains(EAP_DOMAIN)) {
+            return Identifier.JBOSS_EAP_DOMAIN.getName();
+        } else if (Identifier.JBOSS_EAP.getArtifactID().equals(artifactId) && getName().contains(EAP) && !getName().contains(EAP_DOMAIN)) {
+            return Identifier.JBOSS_EAP.getName();
+        } else if (Identifier.WILDFLY_DOMAIN.getArtifactID().equals(artifactId) && getName().contains(DOMAIN)) {
+            return Identifier.WILDFLY_DOMAIN.getName();
+        } else if (Identifier.WILDFLY.getArtifactID().equals(artifactId) && !getName().contains(DOMAIN)) {
+            return Identifier.WILDFLY.getName();
+        } else if (Identifier.GLASSFISH.getArtifactID().equals(artifactId)) {
+            return Identifier.GLASSFISH.getName();
+        } else if (Identifier.PAYARA.getArtifactID().equals(artifactId)) {
+            return Identifier.PAYARA.getName();
+        }
+
+        return "";
+    }
+
+    public String getChameleonTarget(String version) {
+        return getNameForChameleon() + ":" + version + ":" + getContainerType();
+    }
+
+    public boolean isSupportedByChameleon(String version) throws Exception {
+        String containerName = this.getNameForChameleon();
+        if (!containerName.isEmpty()) {
+            String chameleonTarget = getChameleonTarget(version);
+            return Target.from(chameleonTarget).isSupported();
+        }
+
+        return false;
+    }
+
+    // This is work around as method written in chameleon-model is
+    // not working if called from initializeUI of Command AddContainerDependecyStep.
```

Comment:
If we execute this method inside `intializeUI()` then we are getting `unable to call Callable()` exeption.